config/sampler_config.go

```diff
@@ -45,12 +47,379 @@ type RulesBasedSamplerCondition struct {
 	Field    string
 	Operator string
 	Value    interface{}
+	Datatype string
```

Comment:
I expected `DataType` to come into play with the match functions attempted type conversion but I don't see it being used.  Instead it looks like we attempt to convert the incoming data to the Value's data type.  If thats the intention then I think we can remove DataType.

Reply:
I pushed a version with DataType removed.

Reply:
We talked at some length yesterday and Datatype is required because without it, this PR would be a breaking change.